### PR TITLE
fix!: transform valueHasUri values from node to string type (DEV-1047)

### PR DESCRIPTION
**Pull request by mpro7**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: DEV-1047

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: webapi/src/test/scala/org/knora/webapi/store/triplestore/upgrade/plugins/UpgradePluginPR2094Spec.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      model
+        .find(
+          subj = Some(subj),
+          pred = Some(pred),
+          obj = None
+        )
+        .toSet
+        .headOption match {
+        case Some(statement: Statement) =>
+          println(111, statement)
```
Comment (BalduinLandolt) [inline webapi/src/test/scala/org/knora/webapi/store/triplestore/upgrade/plugins/UpgradePluginPR2094Spec.scala:39]: remove print statement

File: test_data/upgrade/pr2094.trig [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+# bad valueHasUri value and missing datatype
+<http://rdfh.ch/0103/5LE8P57nROClWUxEPJhiug/values/fEbt5NzaSe6GnCqKoF4Nhg/standoff/2>
+    rdf:type                                knora-base:StandoffUriTag ;
+    knora-base:standoffTagHasEnd            50 ;
+    knora-base:standoffTagHasStart          20 ;
+    knora-base:standoffTagHasStartIndex     2 ;
+    knora-base:standoffTagHasStartParent     <http://rdfh.ch/0103/5LE8P57nROClWUxEPJhiug/values/fEbt5NzaSe6GnCqKoF4Nhg/standoff/1> ;
+    knora-base:standoffTagHasUUID           "dd2e785b-041e-4bcb-a77d-570efe8a068c" ;
+    knora-base:valueHasUri                  <http://www.maison-george-sand.fr/> .
```
Comment (BalduinLandolt) [inline test_data/upgrade/pr2094.trig:12]: not sure if that makes much sense... but you could consider testing also the case where a URI should not be converted to a string?

**Review by BalduinLandolt — APPROVED** [review]
looks good: you missed some `println()`, and one question regarding the test. But otherwise should be ok

File: webapi/src/main/scala/org/knora/webapi/store/triplestore/upgrade/plugins/UpgradePluginPR2094.scala [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+            statementsToAdd += nodeFactory.makeStatement(
+              subj = statement.subj,
+              pred = statement.pred,
+              obj = newObjectValue(node.iri),
+              context = statement.context
+            )
+
+            log.info(
+              s"Transformed v23 valueHasIri $node to ${newObjectValue(node.iri)}."
```
Comment (irinaschubert) [inline webapi/src/main/scala/org/knora/webapi/store/triplestore/upgrade/plugins/UpgradePluginPR2094.scala:41]:
```suggestion
              s"Transformed v23 valueHasUri $node to ${newObjectValue(node.iri)}."
```

**Review by irinaschubert — APPROVED** [review]
LGTM, spotted just one typo. I assume you've tested it with real standoff data, meaning testing it locally and checking if the app can handle it etc.?

**Review by subotic — APPROVED** [review]
LGTM! you really like to deploy major versions ;-)

**Comment by mpro7** [comment]
@irinaschubert I double checked that and `valueHasUri` value should be represented by the typed string, no matter if it's simple link or standoff link, like below in freshly created value: 
```
<http://rdfh.ch/0001/qN1igiDRSAemBBktbRHn6g/values/D1WBxuH9SFKuh9ACgHwBaw/standoff/9>
        rdf:type                        knora-base:StandoffUriTag ;
        knora-base:standoffTagHasEnd    65 ;
        knora-base:standoffTagHasStart  59 ;
        knora-base:standoffTagHasStartIndex
                9 ;
        knora-base:standoffTagHasStartParent
                <http://rdfh.ch/0001/qN1igiDRSAemBBktbRHn6g/values/D1WBxuH9SFKuh9ACgHwBaw/standoff/8> ;
        knora-base:standoffTagHasUUID   "gjmHSDLTS4Ss-Nqqi_tJ_w" ;
        knora-base:valueHasUri          "http://www.string.ch"^^xsd:anyURI .
```

File: test_data/upgrade/pr2094.trig (hunk 2 above)
Comment (mpro7) [reply test_data/upgrade/pr2094.trig:12]: This plugin is very specific and picks only the node values in type `IriNode`. IMO as with the PR2079 plugin this is enough.

**Pull request merged**